### ISSUE TITLE
Situational adult content labelling reminders

### ISSUE THREAD

**Issue by qazmlp** [bluesky-social/social-app#5628]
### Describe the Feature

If a user regularly posts adult content without self-labelling it (as determined by the moderation service), please nudge them into doing so.

What I would like to see is an informative popup in a supportive tone that explains the feature, why it's important (keeping sexually explicit content away minors, letting users set their own boundaries) _and that the label does not affect visibility in Bluesky's own algorithmic feeds_.

### Describe Alternatives

You could apply the adult content label to accounts who regularly don't self-label porn. I personally would like to avoid that becoming necessary, and if you do end up doing that as a measure of last resort, please do so with much higher threshold than the reminder (and ideally only after manual review, if that doesn't affect your moderators' well-being).

### Additional Context

Too many content creators are used to Twitter and other networks, where a sensitive content label significantly damages their reach, and if applied to their account comes with a search ban. I really want to avoid a similar situation coming about here.

The moderation service here on Bluesky seems to apply labels with some delay, especially on art where the content quite often can't be detected automatically. This means that depending on someone's social graph, a significant amount of nudity and porn can slip through even if they've hidden such content or set it to 'Warn'.

Personally I have set these labels to 'Warn', but I still have to be careful when browsing Bluesky chronologically in public, which makes for an inconvenient user experience.

(This feature request is written with some frustration. I apologise if that shines through.
I appreciate and support adult content creators, I'm just not always in the mood to view their works.)

**Comment by qazmlp** [comment]
This is related to #5383. I think the measures would compliment each other. I think the reminder is necessary even if #5383 is implemented since some creators actually avoid the labels intentionally.

It hasn't happened to me, probably because I'm always super polite and usually indirect when reminding someone to label their posts, but some creators react aggressively when someone asks them to self-label.

**Comment by foxyoreos** [comment]
This would be a large usability improvement.

Getting labels right on Bluesky and not forgetting them is more important because I can't edit labels after the fact. The current UI is so easy to miss. I haven't missed it yet because I'm paranoid about checking, but I can 100% tell you that at some point in the future I will be tired and I will miss the label. It's not a matter of if, it's a matter of when.

If I could set a default label, it would make posting on this network a lot less stressful, because if I post something SFW and accidentally forget to remove the label - oops, but no big deal. If I post something NSFW and forget to add the label, I have to wait for auto-moderation to catch it (if it even does), and in the meantime all of my followers get spammed with unlabeled porn. That's assuming I even notice the mistake.

In the meantime, because tensions on Bluesky are relatively high right now about NSFW art and labeling, I'm much more likely to get added to a bunch of "doesn't label their posts" blocklists if I ever make that mistake. >w<

Basically, this is a foot-gun waiting to happen. A default label that I can override doesn't fix all of the problems, but it fixes a *lot* of them without adding a lot of complexity. Nothing needs to be added to the protocol, it can be a purely clientside feature (although it would be nice to have the setting sync). It would reduce my stress and make the network safer.

**Comment by hvdijk** [comment]
Just today, browsing the What's Hot Classic feed, I've had over 10 unmarked nudes, mostly erect penises, jump right out at me. Of course I report them and block the posters, but it's not just today, it's almost every day, and it makes using Bluesky in public a liability if anyone else sees it. I have previously checked what Bluesky did with reports, and although after a while the posts get flagged, the accounts do not, and their future posts continue to post nudity without any labels indicating so. Please, please start taking this seriously. This issue's suggested option to apply default-labelling of accounts much more swiftly than is done today sounds like the best way of handling it.

A way to set a default label is great for well-meaning posters who accidentally slip up on occasion, it happens, but my personal experience is that the problem is much bigger with posters who do not care at all and would not use such a default label. Dealing with those posters requires moderation involvement.

**Comment by foxyoreos** [comment]
I think the problem is a lot more basic. A lot of critters literally don't know that labeling is a feature.

The UX around labeling is critically bad, and before we assume that posters just don't care, we should first check to see whether they even know the feature exists - whether they're absentmindedly forgetting it, whether they assume that nopony labels content, whether they assume that labeling content affects their position in an algorithm that they think exists.

Active moderation is also going to be necessary, but Bluesky's labeling system is *bad*.

I am obsessive about labeling (I would challenge anypony to find a misslabeled picture on my account). But I have had to delete posts I've made because I missed labels on them. The labeling button is so subtle that you'd mistake it for a "ad" badge on a Google search result. It feels like the site is trying to hide it. Even if you care about labels, you're going to mess up occasionally.

That's not to say that more problematic posters won't exist - they absolutely will. But I am firmly convinced that reminders and default labelers would help with this problem a lot. And I think this issue is smart in that it provides a way to surface reminders to critters who need them. But before we say that critters don't care and this needs to be a moderation crack-down, we should first check if posters know that labels exist, if they know that labels don't limit their reach, if they think that the AI magically handles it for them - I'm not willing to say that default labeling wouldn't be used when default labeling doesn't exist on the site. There's a host of UI problems that contribute to this.

I don't know, maybe it's the communities I'm in, but I don't see critters on Bluesky openly refusing to label posts. I regularly see them asking, "oh, when did Bluesky get labels? I didn't know about them."

**Comment by hvdijk** [comment]
I'm happy to accept that there may be a serious awareness problem. Let me put it differently, without making possibly unwarranted assumptions on the intentions of the posters: the accounts I'm talking about are accounts that post hundreds of porn pictures, every single one of them unlabelled, many of them ending up in the public feeds. Almost every single one of them ends up flagged by the Bluesky Moderation Service, and at no point do these accounts end up either applying labels themselves, nor does the Bluesky Moderation Service end up applying a profile label.

My conclusion from this was that these accounts cannot reasonably not know that they do not need to flag such images, but if your conclusion is that the actions from Bluesky indicate to them the exact opposite, that Bluesky's moderation tells them they *don't* need to flag their own images because Bluesky will do it for them, then something is very wrong, that means the moderation service is in a way harmful by miseducating users.

Ultimately, whether Bluesky needs to improve UX to make them aware before the first image is posted, whether Bluesky needs to improve UX to make them aware after the first image is labelled, whether Bluesky needs to send them a moderation message at some point, whether Bluesky needs to apply a profile label, it makes very little difference. In all cases, it's Bluesky that needs to step in. I'm happy to see whatever is done and see afterwards what the impact of it is.

Consider my previous comment partially withdrawn on the motivation part, then. Still, a way to set a default label, without anything else, would not be sufficient: people who already accounts would have no reason to check their settings to find that new default label setting, so still would not set it. It may be a good improvement, but only if coupled with other UX improvements or changes to moderation.

**Comment by foxyoreos** [comment]
I think what's really good about this feature request is exactly what you're saying - an option doesn't matter if nopony knows that the option is there. And while I (somewhat selfishly) want an option to auto-label posts just in general so that I don't need to be as paranoid about it when I upload, I think on top of that, pairing that with a system that says "you're posting a lot of porn unlabeled, maybe you should have this turned on" is a *great* idea.

> What I would like to see is an informative popup in a supportive tone that explains the feature, why it's important (keeping sexually explicit content away minors, letting users set their own boundaries) and that the label does not affect visibility in Bluesky's own algorithmic feeds.

I think this is an important part of the feature request.

----

> In all cases, it's Bluesky that needs to step in. 

Fully agreed on this. And again, I'll say that this might be a difference in communities, it's possible that the corner of Bluesky that I'm in acts differently.

I understand that Bluesky moderation might be hesitant to start auto-applying labels to profiles since that comes with more consequences (a NSFW label on a profile is essentially a global label that you can't opt out of on any post you make). But that's also why a default labeling system would be great - it lowers those consequences so the moderation could be much faster about turning it on.

> Almost every single one of them ends up flagged by the Bluesky Moderation Service, and at no point do these accounts end up either applying labels themselves, nor does the Bluesky Moderation Service end up applying a profile label.

Also fully agreed that this is kind of ridiculous. The first time that a label gets auto-applied that is an opportunity to at least educate the user that the labeling system exists. Show a pop-up the next time they log in that they can't ignore, something. No one on Bluesky should be under the impression that the AI labeler is what they're supposed to be using, but a lot of critters are, and I understand not wanting to complicate signup, but like.. if somepony uploads hundreds of posts like this and they haven't been told by the site that they're doing something wrong, my take on that is that the site is (intentionally or not) implying to them that what they're doing is correct - it explains a lot to me about why critters might think that the AI labeler is what they're supposed to use if they're relying on it and not getting any warnings or notifications. That is a broken system.
